pkgs/top-level/all-packages.nix

```diff
@@ -9785,38 +9785,40 @@ with pkgs;
   #
   # Android NDK provides an OpenGL implementation, we can just use that.
   #
-  # On macOS, we use the OpenGL framework. Packages that still need GLX
-  # specifically can pull in libGLX instead. If you have a package that
-  # should work without X11 but it can’t find the library, it may help
-  # to add the path to `NIX_CFLAGS_COMPILE`:
+  # On macOS, the SDK provides the OpenGL framework in `stdenv`.
+  # Packages that still need GLX specifically can pull in `libGLX`
+  # instead. If you have a package that should work without X11 but it
+  # can’t find the library, it may help to add the path to
+  # `env.NIX_CFLAGS_COMPILE`:
   #
-  #     -L${libGL}/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
+  #     -L$SDKROOT/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
```

Comment:
Does this work? When the cc-wrapper hook processes the flags, it resolves environment variables?

Reply:
I think that they get expanded inside `env.*`? Maybe with structured attributes you have to do it in a Bash phase instead? I could just recommend that instead.

Reply:
I think it has to be done in Bash in one of the phases. Otherwise, I think you’ll get something like:

```
clang: warning: argument unused during compilation: '-L$SDKROOT/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries' [-Wunused-command-line-argument]
```

Reply:
Amended appropriately. Though since only one package in the tree even does this, probably it wasn’t even worth documenting.